ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -780,8 +786,12 @@ void handleBody(InputStream payload)
         chunkIndexToBuffer = new TreeMap<>();
         if (blobType == BlobType.MetadataBlob) {
           ByteBuffer serializedMetadataContent = blobData.getStream().getByteBuffer();
-          List<StoreKey> keys =
+          MultiPartMetadata metadata =
               MetadataContentSerDe.deserializeMetadataContentRecord(serializedMetadataContent, blobIdFactory);
+          chunkSize = metadata.getChunkSize();
+          totalSize = metadata.getTotalSize();
+          List<StoreKey> keys = metadata.getKeys();
+          //TODO filter keys here to just include blobs in range
```

Comment:
I'll take out this TODO, but it was just a reminder that, for range requests, we only need to fetch the chunks within the range offsets, instead of all chunks.